packages/python/plotly/plotly/express/_core.py

```diff
@@ -322,28 +337,44 @@ def make_trace_kwargs(args, trace_spec, trace_data, mapping_labels, sizeref):
                 if (
                     args["x"]
                     and args["y"]
-                    and len(trace_data[[args["x"], args["y"]]].dropna()) > 1
+                    and len(
+                        trace_data.select(nw.col(args["x"], args["y"])).drop_nulls()
+                    )
+                    > 1
                 ):
                     # sorting is bad but trace_specs with "trendline" have no other attrs
-                    sorted_trace_data = trace_data.sort_values(by=args["x"])
-                    y = sorted_trace_data[args["y"]].values
-                    x = sorted_trace_data[args["x"]].values
+                    sorted_trace_data = trace_data.sort(by=args["x"], nulls_last=True)
+                    y = sorted_trace_data.get_column(args["y"])
+                    x = sorted_trace_data.get_column(args["x"])
 
-                    if x.dtype.type == np.datetime64:
+                    if x.dtype == nw.Datetime:
                         # convert to unix epoch seconds
-                        x = x.astype(np.int64) / 10**9
-                    elif x.dtype.type == np.object_:
+                        x = (
+                            x.to_frame()
+                            .select(
+                                **{
+                                    args["x"]: nw.when(~x.is_null()).then(
+                                        x.cast(nw.Int64())
+                                    )
```

Comment:
Controlling for null because:

```py
pd.to_datetime(['2020', None]).astype('int64')
Index([1577836800000000000, -9223372036854775808], dtype='int64')
```

Reply:
Perhaps we should introduce `.dt.timestamp` in Narwhals and take care of that there?